src/middlewares/apiErrorHandler.ts

```diff
@@ -0,0 +1,15 @@
+import { Request, Response, NextFunction } from 'express';
+
+interface ApiError extends Error {
+  status?: number;
+}
+
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
+export const errorHandler = (err: ApiError, req: Request, res: Response, next: NextFunction) => {
+  console.error(err.stack);
+
+  const status = err.status || 500;
+  const message = err.message || 'An internal server error occurred';
+
+  res.status(status).json({ error: message });
+};
```

Comment:
Maybe I have missed it but where do you define other HTTP errors, such as 404 or 400?

Reply:
Yes, I have implemented structured error handling in another project using predefined ApiError methods. I wish I had incorporated this approach into this project as well. Here is an example of how I have managed errors in the past:`export default class ApiError extends Error {
  constructor(
    readonly statusCode: number,
    readonly message: string,
    readonly source?: Error
  ) {
    super();
  }
}

export class NotFoundError extends ApiError {
  constructor(readonly message: string = "Not Found", source?: Error | any) {
    super(404, message, source);
  }
}

export class ForbiddenError extends ApiError {
  constructor(readonly message: string = "Forbidden", source?: Error | any) {
    super(403, message, source);
  }
}

export class InternalServerError extends ApiError {
  constructor(
    readonly message: string = "Internal Server Error",
    source?: Error | any
  ) {
    super(500, message, source);
  }
}

export class UnauthorizedError extends ApiError {
  constructor(
    readonly message: string = "Unauthorized Request",
    source?: Error | any
  ) {
    super(401, message, source);
  }
}

export class BadRequestError extends ApiError {
  constructor(readonly message: string = "Bad Request", source?: Error | any) {
    super(400, message, source);
  }
}`